manifests/profile/compiler.pp

```diff
@@ -0,0 +1,46 @@
+# @summary Apply this class to a master or compiler to collect puppetserver metrics
+#
+# @param timeout
+#   Deafault timeout of http calls.  Defaults to 5 seconds
+#
+# @param compiler
+#   The FQDN of the compiler / master.  Defaults to the FQDN of the server where the profile is applied
+#
+# @param port
+#   The port that the puppetserver service listens on on your compiler.  Defaults to 8140
+#
+# @param interval
+#   The frequency that telegraf will poll for metrics.  Defaults to '5s'
+define puppet_metrics_dashboard::profile::compiler (
+  String[2] $timeout                               = $puppet_metrics_dashboard::params::http_response_timeout,
+  Variant[String,Tuple[String, Integer]] $compiler = $facts['networking']['fqdn'],
+  Integer[1] $port                                 = 8140,
+  String[2] $interval                              = '5s',
+  ){
+
+  telegraf::input { "puppetserver_metrics_${compiler}":
+    plugin_type => 'httpjson',
+    options     => [{
+      'name'                 => 'puppet_stats',
+      'servers'              => [ "https://${compiler}:${port}/status/v1/services?level=debug" ],
```

Comment:
Assuming this defined type is being applied to the node being queried you can simplify this by calling localhost instead since you are already disabling certificate verification:

```suggestion
      'servers'              => [ "https://localhost:${port}/status/v1/services?level=debug" ],
```

Reply:
These profiles need to work if we apply them on a stand-alone grafana server where telegraf runs and collects all the metrics, as well as on PE nodes, therefore we need to use a hostname so that either method will work.